apps/api-gateway/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,55 @@
+import { Controller, Get, Post, Req, Res, UseGuards } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Request, Response } from 'express';
+import { AuthService } from './auth.service';
+import { GithubGuard } from './guards/github.guard';
+import { JwtRefreshGuard } from './guards/jwt-refresh-auth.guard';
+import { Payload } from './types/jwt.type';
+import { GithubUser } from './types/user.type';
+
+@Controller('auth')
+export class AuthController {
+  public constructor(
+    private readonly authService: AuthService,
+    private readonly configService: ConfigService,
+  ) {}
+
+  @Get('github')
+  @UseGuards(GithubGuard)
+  private async gituhblogin() {}
+
+  @Get('github/callback')
+  @UseGuards(GithubGuard)
+  private async githubLoginCallback(
+    @Req() req: Request,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    const loginUser = await this.authService.login(req.user as GithubUser);
+    const clientBaseUrl = this.configService.get<string>('network.CLIENT_MAIN_URL');
+    const clientSignupUrl = this.configService.get<string>('network.CLIENT_SIGNUP_URL');
+
+    if (loginUser.isExist === false) {
+      res.redirect(301, `${clientSignupUrl}?email=${loginUser.email}&nickname=${loginUser.name}`);
+
+      return;
```

Comment:
url을 이렇게 만들면 휴먼에러의 위험성이 크다고 생각해서요~
요런 방식으로 다뤄줘도 좋을 것 같네요!

```ts
const baseUrl = 'https://example.com/path';
const url = new URL(baseUrl);

url.searchParams.append('name', 'John Doe');
url.searchParams.append('age', '25');

const result = url.toString();
```

Reply:
(취향) 반환값이 없는 함수 실행 종료 목적으로서의 return은 저는 붙이는게 더 깔끔하다고 생각이 들기도 합니다~

```suggestion
      res.redirect(301, `${clientSignupUrl}?email=${loginUser.email}&nickname=${loginUser.name}`);
      return;
```

요것도 괜찮다고 생각하구요!
```suggestion
      return res.redirect(301, `${clientSignupUrl}?email=${loginUser.email}&nickname=${loginUser.name}`);
```

Reply:
[6d6807c](https://github.com/GrebbMe/grabbme-be/pull/53/commits/6d6807c576cffa4bda02e9d5db3e0bd1750726be)
return 과 res.redirect를 붙여 코드 변경 하였습니다!